### PR TITLE
DSG-2793: sort rejected documents

### PR DESCRIPTION
**Pull request by lucywang-nv**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Sort the rejected documents so that it reflects the order in the form.

## Screenshot of the affected changes
Before: 
Before
Before-FinancialBenefit
image

After:
After
After - FinancialBenefit
image

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/public/feature/dashboard/src/lib/services/dashboard.service.ts [hunk 1]
```diff
@@ -49,21 +59,69 @@ export class DashboardService {
...
     );
   }
 
+  private _getDocumentOrder(transactionModel: TransactionModel): string[] {
+    const documentAttributes: string[] = [];
+
+    if (!this._documentOrder.has(transactionModel.transactionDefinitionKey)) {
+      const formComponents = this._transactionDefinitionFormModelMap.get(transactionModel.transactionDefinitionKey);
+      const docComponents = formComponents?.components
+        .filter(comp => comp.key === 'attachments' || comp.key === 'documents')
```
Comment (kevincheng0) [inline libs/public/feature/dashboard/src/lib/services/dashboard.service.ts:85]: The component key doesn't determine the type of that component. For example, a transaction can have a component with the key 'identification', that contains various file upload components. In this case, 'attachments' and 'documents' are the keys that we chose for FinancialBenefit or UnemploymentInsurance. They way to identify the type of a component is if the `type` property is `nuverialFileUpload` or `nuverialMultipleFileUpload`. DFS through the transaction model can work, and you can probably get the key of each component along the way. Or maybe a deep filter and flatten.

File: libs/public/feature/dashboard/src/lib/services/dashboard.service.ts [hunk 2]
```diff
@@ -49,21 +59,69 @@ export class DashboardService {
...
+    if (docComponent.components) {
+      for (const component of docComponent.components) {
+        this._getDocumentKey(component, documentAttributes);
+      }
+    } else {
+      docComponent.key && documentAttributes.push(docComponent.key);
+    }
+  }
+
+  private _sortRejectedDocuments(sortOrder: string[], firstDoc: RejectedDocument, secondDoc: RejectedDocument) {
```
Comment (kevincheng0) [inline libs/public/feature/dashboard/src/lib/services/dashboard.service.ts:114]: We should probably consider the order of documents in a multiple document upload component. Each `RejectedDocument` has an `index` property, and we should compare those as well.

**Review by kevincheng0 — CHANGES_REQUESTED** [review]
Looks good, just some changes in order to make it general to any transaction

File: libs/public/feature/dashboard/src/lib/services/dashboard.service.ts (hunk 1 above)
Comment (lucywang-nv) [reply libs/public/feature/dashboard/src/lib/services/dashboard.service.ts]: Made the change, using DFS.

File: libs/public/feature/dashboard/src/lib/services/dashboard.service.ts (hunk 2 above)
Comment (lucywang-nv) [reply libs/public/feature/dashboard/src/lib/services/dashboard.service.ts:107]: Changed.

**Review by kevincheng0 — APPROVED** [review]
Looks good :)

**Review by davidyoung77 — APPROVED** [review]

**Pull request merged**